magenta/music/musicxml_parser.py

```diff
@@ -178,24 +178,20 @@ def get_score(filename):
   def _parse(self):
     """Parse the uncompressed MusicXML document."""
     # Parse part-list
-    xml_part_list = self.score.find('part-list')
-    for element in xml_part_list:
-      if element.tag == 'score-part':
-        score_part = ScorePart(self.state, element)
-        self.score_parts.append(score_part)
+    xml_part_list = self._score.find('part-list')
+    if xml_part_list is not None:
+      for element in xml_part_list:
+        if element.tag == 'score-part':
+          score_part = ScorePart(element)
+          self._score_parts[score_part.id] = score_part
 
     # Parse parts
-    for score_part_index, child in enumerate(self.score.findall('part')):
-      # If a score part is missing, add a default score part
-      if score_part_index >= len(self.score_parts):
-        score_part = ScorePart(self.state)
-        self.score_parts.append(score_part)
-
-      part = Part(child, self.score_parts[score_part_index], self.state)
+    for score_part_index, child in enumerate(self._score.findall('part')):
```

Comment:
There is supposed to be a 1:1 between ScorePart (MIDI info like track, patch, etc.) and Part, but I have found some (particularly older) MusicXML files where this not true. If a Part is found but there isn't a ScorePart, the parser creates a default ScorePart with default MIDI patch. The score_part_index then is used to associate the correct ScorePart with the Part.

I know this is confusing, and I would have preferred it if the MIDI info was contained directly in the Part info and that there wasn't a ScorePart element. I'm open to suggestions on better ways to handle this. For example, we may want to raise an Exception if there is a missing ScorePart, rather than try to correct bad data. This might simplify the code, but I do have a lot of older MusicXML that seem to have this encoding error.